### PR TITLE
Translate most of git.txt to german

### PR DESCRIPTION
**Pull request by rimrul**
I feel like the approach in da557f8 is a bit unorthodox and it causes some hiccups on making the project. Any ideas for a better approach to this?

### CODE REVIEW

File: po/documentation.de.po [hunk 1]
```diff
@@ -8,8 +8,13 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: git documentation\n"
+<<<<<<< HEAD
 "POT-Creation-Date: 2019-01-26 17:40+0100\n"
 "PO-Revision-Date: 2019-01-16 13:41+0100\n"
+=======
```
Comment (rimrul) [inline po/documentation.de.po]: How did these get past the mergetool? Idk. Well, I'll have to push an update anyways.

Comment (rimrul) [reply po/documentation.de.po]: Ok, I've messed this up, again. Great.

Comment (jnavila) [reply po/documentation.de.po]: This is due to the wildcards in the Makefile:

```Makefile
LANGUAGE_PO = $(wildcard po/*.po)
```

That catches your special po file.

**Comment by rimrul** [comment]
Yes, I thought so. Maybe I should just rename the file to .txt, since it isn't really a .po file anyways . Or maybe a gettext compendium is the proper way to do this. Or maybe something like the "translation memory" feature in lokalize or the "glossary" in Weblate is the right way to do this. I don't really know how these things are usually handled. But I somehow doubt we're the first project with an issue like this.

**Comment by jnavila** [comment]
I had quite a similar approach in mind and your solution is elegant and reusable across languages. So, maybe just restricting the expansion in the Makefile should be enough.

**Comment by jnavila** [comment]
> I don't really know how these things are usually handled. But I somehow doubt we're the first project with an issue like this.

I guess most translators are not regex experts and don't use this kind of trick. So, we may as well be the first project to use regex. This is also applicable to long options with parameters such as `--foo=<bar>`, where we would like to translate `bar`.

**Comment by rimrul** [comment]
> I guess most translators are not regex experts and don't use this kind of trick. So, we may as well be the first project to use regex.
 
Yes, but I've thought even if the solution isn't common the problem might be. And if the problem is common, there's probably a standard solution. I'm fine with keeping it like this though. This concept should be fairly extensible when needed and not too hard to explain to new translators.
Anyways, I've update da557f80 to a9721f8 to exclude the special po file from the Makefile and drop some trailing whitespaces.

**Comment by jnavila** [comment]
Good. Just ping me when it's ready for merging.

**Comment by rimrul** [comment]
Will do. Btw, weblate still seems to ignore the `ignore-same` tags. You might need to run "Update quality checks".

File: po/documentation.de.po [hunk 2]
```diff
@@ -1878,7 +1878,7 @@ msgstr "Zwischen doppelten Anführungszeichen müssen doppelte Anführungszeiche
 #. type: Plain text
 #: en/config.txt:80
 msgid "The following escape sequences (beside `\\\"` and `\\\\`) are recognized: `\\n` for newline character (NL), `\\t` for horizontal tabulation (HT, TAB)  and `\\b` for backspace (BS).  Other char escape sequences (including octal escape sequences) are invalid."
-msgstr "Die folgenden Escape-Sequenzen (neben '\\\"' und '\\\\') werden erkannt: '\\n' als Zeilenumbruch (NL), '\\t' für Tab (HT, TAB) und '\\b' für Backspace (BS). Andere Escape-Sequenzen (auch oktale Escape-Sequenzen) sind ungültig."
+msgstr "Die folgenden Escape-Sequenzen (neben '\\\"' und '\\\\') werden erkannt: '\\n' als Zeilenumbruch (NL), '\\t' für Tab (HT, TAB) und '\\b' für Backspace (BS). Andere Escape-Sequenzen (auch oktale Escape-Sequenzen) sind ungültig.ungültig."
```
Comment (rimrul) [inline po/documentation.de.po]: I've definitely messed something up here.

File: po/documentation.de.po [hunk 3]
```diff
@@ -14572,10 +14570,9 @@ msgstr ""
 
 #. type: Labeled list
 #: en/git-credential-store.txt:32
-#, fuzzy, no-wrap
-#| msgid "--file=<file>"
+#, no-wrap
 msgid "--file=<path>"
-msgstr "--file=<file>"
+msgstr "--file=<Pfad>"
```
Comment (rimrul) [inline po/documentation.de.po]: TODO: s/Pfad/Datei

**Comment by rimrul** [comment]
92.44%. we're getting close.

**Comment by rimrul** [comment]
I think it's ready. As discussed, it isn't quite complete, but 94.6% is good enough for now.

**Comment by rimrul** [comment]
I've messed up and caused a weblate merge conflict. Will push a fix in a couple hours.

**Pull request merged**